core/controllers/base_test.py

```diff
@@ -593,3 +595,64 @@ def test_get_items(self):
                 'param2=value%20with%20%26%20%2B%20-%20/&'
                 'param3=value%20with%20.%20%%20@%20123%20=%20!%20%3C%3E')
             self.assertDictContainsSubset(params, result)
+
+
+class ControllerClassNameTest(test_utils.GenericTestBase):
+
+    def test_controller_class_names(self):
+        """This function checks that all controller class names end with
+        either 'Handler' or 'Page'.
+        """
+        for url in main.URLS:
+            if not isinstance(url, tuple):
+                handler = str(url.handler)[8:-2].rsplit('.', 1)[1]
```

Comment:
Why do all this stuff to find the class when you already have a direct reference to the class in main.py?